ygot/struct_validation_map.go

```diff
@@ -332,18 +332,35 @@ func MergeJSON(a, b map[string]interface{}) (map[string]interface{}, error) {
 	return o, nil
 }
 
+// MergeOption is an interface that is implemented by all parameters which specify a merge
+// behaviour when merging GoStructs.
+type MergeOption interface {
+	// IsMergeOption is a marker method for these options.
+	IsMergeOption()
+}
+
+// AllowLeafOverwrite allows scalar (leaf) fields to be overwritten in the destination
+// strucft when specified.
+type AllowLeafOverwrite struct {
+	IfEqual bool // IfEqual allows leaves to be overwritten only if their value is equal.
+}
```

Comment:
This is a bit confusing. Why would you not allow an override if values are equal?

Reply:
It's the other way around -- explicitly overwriting is disabled if `a` and `b` both have a populated field, but that field is not equal (i.e., there are two list entries that have the same map key, but their `Name` field - using the above example is not equal) then this throws an error.